gpu-service/jobs.py

```
@@ -0,0 +1,462 @@
+def process_audio_upload(bucket_name, object_name, event_data=None):
+    """
+    RQ job function to process audio files for transcription.
+    """
+    try:
+        # Ensure transcription directory exists
+        os.makedirs("/app/transcriptions", exist_ok=True)
+
+        # Download audio from MinIO
+        audio_obj = minio_client.get_object(bucket_name, object_name)
+        audio_bytes = audio_obj.read()
+        audio_path = tempfile.mktemp(suffix=os.path.splitext(object_name)[1])
```

Ruff (0.11.9)

Use of insecure and deprecated function (`mktemp`)

(S306)